teku/src/main/java/tech/pegasys/teku/cli/subcommand/ValidatorApiClientUtil.java

```
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.cli.subcommand;
+
+import static tech.pegasys.teku.infrastructure.logging.SubCommandLogger.SUB_COMMAND_LOG;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpClientAuthLoggingIntercepter;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpValidatorRestApiClient;
+
+class ValidatorApiClientUtil {
+  static Spec getSpec(OkHttpValidatorRestApiClient apiClient) {
+    return apiClient
+        .getConfigSpec()
+        .map(response -> SpecConfigLoader.loadConfig(response.data))
+        .map(SpecFactory::create)
+        .orElseThrow();
```

nit: We'll probably want to make this throw `InvalidConfigurationException` with a useful error message if the spec can't be loaded.  That should cause the client to exit and print the message cleanly to the user.